src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -191,36 +191,38 @@ void Channel::JoinChannel(Player* player, std::string const& pass)
 
     player->JoinedChannel(this);
 
-    if (_announce && !player->GetSession()->HasPermission(rbac::RBAC_PERM_SILENTLY_JOIN_CHANNEL))
+    if (_announceEnabled && !player->GetSession()->HasPermission(rbac::RBAC_PERM_SILENTLY_JOIN_CHANNEL))
     {
         WorldPacket data;
         MakeJoined(&data, guid);
         SendToAll(&data);
     }
 
-    PlayerInfo pinfo;
-    pinfo.player = guid;
+    bool newChannel = _playersStore.empty();
+
+    PlayerInfo& pinfo = _playersStore[guid];
     pinfo.flags = MEMBER_FLAG_NONE;
-    playersStore[guid] = pinfo;
+    pinfo.invisible = !player->isGMVisible();
 
     WorldPacket data;
     MakeYouJoined(&data);
     SendToOne(&data, guid);
 
-    JoinNotify(guid);
```

Comment:
by looking at this PR I have no clue if this line was removed on purpose (whatever the purpose would be) or if it was removed by accident